src/File.php

```diff
@@ -1536,4 +1536,14 @@ protected function getFilter()
     {
         return FileNameFilter::create();
     }
+
+    /**
+     * Get an array with each segment (directory name and file name) in a file path as a value.
+     */
+    protected static function getFilePathParts(string $filePath): array
+    {
+        // Explicitly allow zero as a segment in the path (e.g. /some/path/0/file.txt)
+        $notEmpty = fn(mixed $part) => ($part === 0 || $part === '0' || (bool) $part);
+        return array_filter(preg_split("#[/\\\\]+#", $filePath), $notEmpty);
```

Comment:
```suggestion
        return array_values(array_filter(preg_split("#[/\\\\]+#", $filePath), 'strlen'));
```
And get rid of the $notEmpty callable above

Note I've added array_values() to reindex the array, otherwise "/lorem/ipsum/my.txt" will have its first index as 1

Reply:
This PR is only trying to add `0` and `'0'` as valid values where they weren't previously.
Any changes to indexes were already there, and I don't want to mess with them in this PR.
Any type wonkiness was already there, and I don't want to mess with it in this PR.